src/components/VideoPage/VideoPage.js

```diff
@@ -0,0 +1,124 @@
+import { useState, useContext, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { RiThumbUpFill, RiPlayListAddFill } from "react-icons/ri";
+import { IoIosShareAlt } from "react-icons/io";
+
+import VideoCard from "../VideoCard/VideoCard";
+import Modal from "../Modal/Modal";
+import PlaylistForm from "../Playlist/PlaylistForm/PlaylistForm";
+import { UserContext } from "../../store/UserContext/UserContext";
+
+import * as userActionTypes from "../../store/types/userActionTypes";
+import { showToast, convertToShortNumber } from "../../utils";
+import { videos } from "../../data";
+
+import "./video-page.css";
+
+export default function VideoPage() {
+  const { user, userDispatch } = useContext(UserContext);
+
+  const { embedId } = useParams();
+
+  const [isPlaylistModalOpen, setIsPlaylistModalOpen] = useState(false);
+
+  const togglePlaylistModal = () => {
+    setIsPlaylistModalOpen(!isPlaylistModalOpen);
+  };
+
+  const copyUrlHandler = () => {
+    navigator.clipboard.writeText(window.location.href);
+    showToast("Link copied to clipboard!");
+  };
+
+  useEffect(() => {
+    userDispatch({
+      type: userActionTypes.UPDATE_HISTORY,
+      payload: {
+        videoEmbedId: embedId,
+      },
+    });
+    window.scroll({
+      top: 0,
+      left: 0,
+      behavior: "smooth",
+    });
+  }, [embedId]);
+
+  const { title, views, age, channel, likes } = videos.find(
+    (video) => video.embedId === embedId
+  );
+
+  const isVideoLiked = user.likedVideos.includes(embedId);
+
+  const videoLikeHandler = () => {
+    userDispatch({
+      type: userActionTypes.TOGGLE_LIKED_VIDEO,
+      payload: {
+        videoEmbedId: embedId,
+        isVideoAlreadyLiked: isVideoLiked,
+      },
+    });
+  };
+
+  return (
+    <div className="video-page-container">
+      <div className="video-page-left">
+        <div className="video-wrapper">
+          <iframe
+            className="iframe-video"
+            src={"https://www.youtube.com/embed/" + embedId + "?enablejsapi"}
```

Comment:
Would like to know why you have appended "?enablejsapi" ahead of your id?

Reply:
This query does not have any impact now and I'll remove it. But the purpose is to take control of the video using JS.